### PR TITLE
Angular 17 update

### PR DESCRIPTION
**Pull request by ectuser**
Update angular version and some docs

### CODE REVIEW

**Comment by ectuser** [comment]
Hi @czeckd! I've prepared this small MR with angular upgrade. I've also tested it locally and it worked well.

**Comment by sheikalthaf** [comment]
@ectuser  also we can update to [application builder](https://angular.dev/tools/cli/esbuild#using-the-application-builder) as described here. Because it is going to be the default one in future. FYI @czeckd

**Comment by czeckd** [comment]
Thank you @ectuser! Would you be able to add the application builder changes too to your pull request please?

**Comment by ectuser** [comment]
Hi @sheikalthaf @czeckd , the application builder is updated. However it's not yet available for library builder. But for now, it's done :)

File: angular.json [hunk 1]
```diff
@@ -33,11 +33,11 @@
...
       "architect": {
         "build": {
-          "builder": "@angular-devkit/build-angular:browser",
+          "builder": "@angular-devkit/build-angular:application",
           "options": {
             "outputPath": "dist/svg-icon",
             "index": "src/index.html",
-            "main": "src/main.ts",
+            "browser": "src/main.ts",
             "polyfills": "src/polyfills.ts",
```
Comment (sheikalthaf) [inline angular.json]: It should be an array for application builder

**Review by sheikalthaf — CHANGES_REQUESTED** [review]

File: angular.json (hunk 1 above)
Comment (czeckd) [reply angular.json]: @ectuser - Looks like it needs to be an array. After building the distribution, here's the error:

```
npm start

> svg-icon@17.0.0 start
> ng serve --configuration production

Node.js version v21.1.0 detected.
Odd numbered Node.js versions will not enter LTS status and should not be used for production. For more information, please see https://nodejs.org/en/about/previous-releases/.
Option "browserTarget" is deprecated: Use 'buildTarget' instead.
Error: Schema validation failed with the following errors:
  Data path "/polyfills" must be array.
```

Comment (ectuser) [reply angular.json]: Yes, you're right, added in the https://github.com/czeckd/angular-svg-icon/pull/150/commits/03dd59dc9097b60312cba51313872314113ffbc9

Also renamed browserTarget property to buildTarget.

**Comment by sheikalthaf** [comment]
LGTM

**Comment by czeckd** [comment]
@ectuser thanks for the pull! @sheikalthaf thanks for the review! Getting this merged now.

**Pull request merged**